diff --git a/hw/display/vga.c b/hw/display/vga.c
index 40adeb3e2f..c13b1cc6ae 100644
--- a/hw/display/vga.c
+++ b/hw/display/vga.c
@@ -2254,6 +2254,8 @@ bool vga_common_init(VGACommonState *s, Object *obj, Error **errp)
     vmstate_register_ram(&s->vram, s->global_vmstate ? NULL : DEVICE(obj));
     xen_register_framebuffer(&s->vram);
     s->vram_ptr = memory_region_get_ram_ptr(&s->vram);
+    /* DMA hole: preallocate the ram */
+    memset(s->vram_ptr, 0, s->vram_size);
     s->get_bpp = vga_get_bpp;
     s->get_params = vga_get_params;
     s->get_resolution = vga_get_resolution;
